src/main/kotlin/com/amazon/ionelement/impl/IonElementLoaderImpl.kt

```diff
@@ -128,26 +142,41 @@ internal class IonElementLoaderImpl(private val options: IonElementLoaderOptions
                     IonType.BLOB -> BlobElementImpl(ionReader.newBytes(), annotations, metas)
                     IonType.LIST -> {
                         ionReader.stepIn()
-                        val list = ListElementImpl(loadAllElements(ionReader).toImmutableListUnsafe(), annotations, metas)
+                        val listContent = ArrayList<AnyElement>()
```

Comment:
I wonder if it would improve performance to pool these using something like IonJava's RecyclingStack. We could get away with having one per depth level right?

Reply:
I'm intentionally not using a recycling stack because on L154,  we wrap it in an immutable list, and then forget about the underlying mutable reference.
```kotlin
ListElementImpl(listContent.toImmutableListUnsafe(), annotations, metas)
```

However, it would be worth investigating whether there is a performance benefit to having a pool of ArrayLists that have a relatively larger _capacity_, and then copying to an exact-sized immutable list.